docs/src/main/asciidoc/grpc-getting-started.adoc

```diff
@@ -0,0 +1,318 @@
+    }
+}
+
+----
+1. Expose your implementation as bean
+2. Extends the `ImplBase` class. This is a generated class.
+3. Implement the methods defined in the service definition (here we have a single method)
+4. Build and send the response
+5. Close the response
+
+Quarkus also provides an additional model with Mutiny.
+A Mutiny implementation of this service would be:
+
+[source, java]
+----
+package org.acme;
+
+import io.quarkus.example.HelloReply;
+import io.quarkus.example.HelloRequest;
+import io.quarkus.example.MutinyGreeterGrpc;
+import io.smallrye.mutiny.Uni;
+
+import javax.inject.Singleton;
+
+@Singleton
+public class ReactiveHelloService extends MutinyGreeterGrpc.GreeterImplBase {
+
+    @Override
+    public Uni<HelloReply> sayHello(HelloRequest request) {
+        return Uni.createFrom().item(() ->
+                HelloReply.newBuilder().setMessage("Hello " + request.getName()).build()
+        );
+    }
+}
+----
+
+The main differences are the following:
+
+* it extends the `ImplBase` from `MutinyGreeterGrpc` instead of `GreeterGrpc`
+* the signature of the method is using Mutiny types
+
+=== The gRPC server
+
+The services are _served_ by a _server_.
+Available services (_CDI beans_) are automatically registered and exposed.
+
+By default, the server is exposed on `localhost:9000`, and uses _plain-text_ (so no TLS).
+
+Run the application using: `mvn quarkus:dev`.
+
+== Consuming a gRPC service
+
+In this section, we are going to consume the service we expose.
+To simplify, we are going to consume the service from the same application, which in the real world, does not make sense.
+
+Open the existing `org.acme.ExampleResource` class, and edit the content to become:
+
+[source, java]
+----
+package org.acme;
+
+import io.quarkus.example.GreeterGrpc;
+import io.quarkus.example.HelloRequest;
+import io.quarkus.grpc.runtime.annotations.GrpcService;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+@Path("/hello")
+public class ExampleResource {
+
+    @Inject
+    @GrpcService("hello")                     // <1>
+    GreeterGrpc.GreeterBlockingStub client;   // <2>
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String hello() {
+        return "hello";
+    }
+
+    @GET
+    @Path("/{name}")
+    public String hello(@PathParam("name") String name) {
+        return client.sayHello(HelloRequest.newBuilder().setName(name).build()).getMessage();  // <3>
+    }
+}
+----
+1. Inject the service and configure its name. This name is used in the application configuration
+2. Use the _blocking_ stub (also a generated class)
+3. Invoke the service
+
+We need to configure the application to indicate where is the `hello` service.
+In the `src/main/resources/application.properties` file, add the following property:
+
+[source, text]
+----
+hello.host=localhost
```

Comment:
How is this property mapped? If it's some magic Quarkus config, I would prefer we have some proper `quarkus.` config.

Reply:
I will have a look tomorrow. When I tried it was not working because it was all made at runtime. But we revisited that, so should work now.

Reply:
Done.